### DISCUSSION TITLE
[question] mouse pointer

### DISCUSSION THREAD

**Discussion by ghost** [ant-design/ant-design#51856] [Q&A] [answered]
Hello everyone,

I'm pretty new to antd, so forgive me if I missed something. I would like in antd5 to change the mouse pointer, in a custom way, when the mouse hover a component. Is there a way to achieve this using either styles or the configuration provider using themes?

Best regards

Xander

**Comment by aojunhao123** [comment, marked as answer]
![QQ_1733131157284](https://github.com/user-attachments/assets/2b929cce-abb7-42b2-b837-0bdcef1746a9)

The `ConfigProvider` does not provide a design token for the mouse pointer, but we can use it to add custom styles or override the internal styles of components. Below is an example using the `Button` component from Ant Design, with the code as follows.

```tsx
import React from "react";
import "./index.css";
import { Button, Flex, ConfigProvider } from "antd";

const App: React.FC = () => (
  <ConfigProvider button={{ className: "my-button" }}>
    <Flex gap="small" wrap>
      <Button type="primary">Primary Button</Button>
      <Button>Default Button</Button>
      <Button type="dashed">Dashed Button</Button>
      <Button type="text">Text Button</Button>
      <Button type="link">Link Button</Button>
    </Flex>
  </ConfigProvider>
);

export default App;
```

**Comment by aojunhao123** [comment]
If you want to see more details about theme customization, you can refer to this official blog: [Extends Theme](https://ant-design.antgroup.com/docs/blog/config-provider-style)

**Comment by ghost** [comment]
Hello @aojunhao123,

Thank you for your quick response. Your solution is promising. Though, I tried to apply your example (combined with the code in extends theme) to get a tag to have such a cursor. I tried the following ([test code can be found here](https://playcode.io/2177485)):

```Javascript
import React from 'react';
import { ConfigProvider, Popover, Tag } from 'antd';
import { createStyles } from 'antd-style';

export default function App() {
  const customMouseOverStyle = () => {
    const { getPrefixCls } = React.useContext(ConfigProvider.ConfigContext);
    const tagPrefixCls = getPrefixCls('tag');
    console.log(tagPrefixCls);

    return createStyles(({ css }) => ({
      tag: css`
        .${tagPrefixCls} {
          cursor: not-allowed;
          border: 1px solid white;
        }
      `,
    }))();
  };

  const { styles } = customMouseOverStyle();
  console.log(styles);

  return (
    <div className='App'>
      <ConfigProvider tag={{ className: styles.tag }}>
        <Tag color='#AA3366'>some-tag</Tag>
      </ConfigProvider>
    </div>
  );
}

// Log to console
console.log('Hello console');
```

Upon inspection, I noticed the new style appears in the head of the page and the tag is added to the span:
![image](https://github.com/user-attachments/assets/05cc67af-cd6f-4505-aa9a-278bfb59ce1b)

But it doesn't seem to work

There is also no mention of it in the styles-inspector of chrome.
![image](https://github.com/user-attachments/assets/3897b5ec-f81b-46b4-9a19-ccdd8c5906cf)

Am I doing something wrong, or is it not supposed to work this way?

Thanx in advance

Xander

**Comment by aojunhao123** [comment]
The selector syntax is incorrect because:
1. The createStyles function already generates a unique className
2. Adding `.${tagPrefixCls}` creates a descendant selector, which looks for a child element with that class
3. Instead, the styles should be applied directly to the element, or use '&' for compound selectors

Correct usage:
![QQ_1733160810822](https://github.com/user-attachments/assets/6e369c33-3306-4388-ada8-9866f2672cd7)

**Comment by ghost** [comment]
Thank you very much for showing me the difference between the example version from the blog and my version, and the mistake I made. Best regards Xander
